### PR TITLE
Enable compilation under jdk 11

### PR DESCRIPTION
**Pull request by yuzhu**
Tested to compile locally under jdk11

### CODE REVIEW

**Comment by yuzhu** [comment]
jenkins, test this please

**Comment by yuzhu** [comment]
jenkins, test this please

**Comment by yuzhu** [comment]
jenkins, test this please

**Comment by yuzhu** [comment]
jenkins, test this please

**Comment by yuzhu** [comment]
jenkins, test this please

**Comment by yuzhu** [comment]
@calvinjia @apc999 PTAL at the change to bump default hdfs client to 3.3.0

File: underfs/hdfs/pom.xml [hunk 1]
```diff
@@ -85,13 +85,10 @@
       </build>
     </profile>
 
-    <!-- Profile to build this UFS module connecting to hdfs 2, this is the default profile -->
+    <!-- Profile to build this UFS module connecting to hdfs 2 -->
```
Comment (calvinjia) [inline underfs/hdfs/pom.xml:88]: Can we add enforcement for Java 8 if it is required by hdfs 3.3 and under?

Comment (yuzhu) [reply underfs/hdfs/pom.xml:88]: i don't think this is necessary, because for hdfs under 3.3, we can still use Java 11 to build to target 8.

**Comment by yuzhu** [comment]
@apc999 @calvinjia anymore comments? I'd like to have this merged before today so nightly tests have a chance to run on this.

**Comment by yuzhu** [comment]
@apc999 @calvinjia  ping?

**Comment by yuzhu** [comment]
jenkins, test this please

**Comment by yuzhu** [comment]
jenkins, test this please

**Comment by calvinjia** [comment]
alluxio-bot, merge this please.

**Pull request merged**